### ISSUE TITLE
Modals are not removed after close (since 1.3.0)

### ISSUE THREAD

**Issue by Manduro** [driftyco/ionic#6378]
#### Short description of the problem:
When an Ionic modal is closed, it still remains in the DOM. This behavior can be seen in the Ionic Codepen 'Open Modal with Form' by inspecting it after closing the modal.
http://codepen.io/ionic/pen/VLwLOG

#### What behavior are you expecting?
In 1.2.4, a modal was correctly removed from the DOM after closing it.

**Which Ionic Version?** 1.x

#### Plunker that shows an example of your issue
http://codepen.io/ionic/pen/VLwLOG

**Comment by danbucholtz** [comment]
Hi @manduro,

I'll take a look at this.  I think this may have come in via a PR.  This should be in 1.3.1 later this week.

Thanks,
Dan

**Comment by Manduro** [comment]
Great, thanks. Probably #6160.

**Comment by danbucholtz** [comment]
Hi @manduro 😄 ,

Looking at the codepen, I don't see the modal being removed when calling `hide` as the codepen shows.

If you want to remove the modal from the dom and destroy it, you need to call `remove`.

Thanks for contributing and choosing Ionic!

Thanks,
Dan

**Comment by Manduro** [comment]
Hi @danbucholtz,

You are correct, I was mistaken. The Codepen does not demonstrate the issue correctly. Something has changed though in the handling of closing a modal since 1.3.0.

Previously I was using this in a directive:
```
modalScope.closeModal = function() {
  modalScope.modal.hide();
};
 
modalScope.$on('modal.hidden', function(thisModal) {
  thisModal.currentScope.modal.remove();
  thisModal.currentScope.$destroy();
});
```

In 1.2.4 the modal closing animation finished first, followed by the removal of the modal and destroying of the scope.

Since 1.3.0 the modal is removed instantly without the animation finishing first. Rewriting it to this makes it work as in 1.2.4 again:
```
modalScope.closeModal = function() {
  modalScope.modal.remove().then(function() {
    modalScope.$destroy();
  });
};
```

Maybe something was fixed in 1.3.0 and I was just doing it the wrong way before and doing it the right way now?

It still seems to me though, that the first method should work as well, but it doesn't anymore.
